### PR TITLE
SearchBar list component, redux fixed, refresh searchBar Component

### PR DESCRIPTION
**Pull request by mariasaavega**
separate search bar and list component, update the reducer filter and when returning to the search bar the previous search no longer appears

### CODE REVIEW

**Comment by mariasaavega** [comment]
separate search bar and list component, update the reducer filter and when returning to the search bar the previous search no longer appears

File: components/SearchBar/index.jsx [hunk 1]
```diff
@@ -22,9 +24,24 @@ const AppWrapper = (navigation) => {
   const filteredList = useSelector((state) => state.userReducer);
   const dispatch = useDispatch();
 
+  const inputValue = React.useRef(null);
```
Comment (camilovegag) [inline components/SearchBar/index.jsx:27]: Why not importing useRef from react, such as { useEffect }

File: components/SearchBar/index.jsx [hunk 2]
```diff
@@ -44,50 +61,7 @@ const AppWrapper = (navigation) => {
       {filteredList["text"].length === 0 ? (
         console.log("No results")
```
Comment (camilovegag) [inline components/SearchBar/index.jsx:62]: Can you show a No results found Text underneath the search bar instead?

File: components/SearchBar/index.jsx [hunk 3]
```diff
@@ -22,9 +24,24 @@ const AppWrapper = (navigation) => {
...
+
+  useEffect(() => {
+    filteredList['text']=[];
+    inputValue.current.clear();
+
+    if(isFocused){ 
+      getInitialData();
+  }  }, [isFocused]) 
+
+  const getInitialData = async () => {}
```
Comment (camilovegag) [inline components/SearchBar/index.jsx:38]: What is this function doing?

File: components/SearchList/index.jsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  Text
+} from "native-base";
+
+export default function SearchList(props)  { 
+return(
+    <FlatList
+    nestedScrollEnabled
+    value=""
+    keyExtractor={(array) => array.profile.email}
+    data={props.list}
```
Comment (camilovegag) [inline components/SearchList/index.jsx:20]: You can destructure `list` from `SearchList({ list })`, avoiding the use of props keyword

File: components/SearchList/index.jsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    renderItem={({ item }) => {
+    return (
+        <Button
+        backgroundColor= "#333"
+        width= {300}
+        alignItems="flex-start"
+        justifyContent="flex-start"
+        onPress={() =>
+            props.navigate("UserResult",{
```
Comment (camilovegag) [inline components/SearchList/index.jsx:29]: Same here, `props.navigate` can be replaced with `navigation.navigate`. Your function may look like this: 
`SearchList({ list, navigation })`

File: components/SearchBar/index.jsx [hunk 6]
```diff
@@ -44,50 +61,7 @@ const AppWrapper = (navigation) => {
...
-                    <Text fontSize="12px" color="white">
-                      {item.profile.email}
-                    </Text>
-                  </Box>
-                </Flex>
-              </Button>
-            );
-          }}
-        />
+        <SearchList list = {filteredList["text"]} {...navigation} />
```
Comment (camilovegag) [inline components/SearchBar/index.jsx:64]: The `navigation` prop is passed down to components by default, so I think you can remove `{...navigation}` and SearchList will be able to access it.

File: components/SearchBar/index.jsx [hunk 7]
```diff
@@ -13,6 +13,8 @@ import {
 import { useSelector, useDispatch } from "react-redux";
 import { filterByName } from "../../redux/actions";
 import { MaterialIcons } from "@expo/vector-icons";
+import { useIsFocused } from "@react-navigation/native";
+import SearchList from "../SearchList";
 
 export default function SearchBar(navigation) {
   return <AppWrapper {...navigation} />;
```
Comment (camilovegag) [inline components/SearchBar/index.jsx:20]: Maybe the `<AppWrapper />` isn't necessary if you destructure `{ navigation }` on the actual component?

**Review by camilovegag — CHANGES_REQUESTED** [review]
I left some comments!

File: components/SearchBar/index.jsx (hunk 6 above)
Comment (mariasaavega) [reply components/SearchBar/index.jsx:64]: When removed, it stops working because it is not passed by default

**Pull request closed without merging**